saleor/core/tasks.py

```diff
@@ -16,18 +21,15 @@ def delete_from_storage_task(path):
 @app.task
 def delete_event_payloads_task():
     delete_period = timezone.now() - settings.EVENT_PAYLOAD_DELETE_PERIOD
-    deliveries = EventDelivery.objects.filter(created_at__lte=delete_period)
-    attempts = EventDeliveryAttempt.objects.filter(
-        Q(Exists(deliveries.filter(id=OuterRef("delivery_id"))))
-        | Q(delivery__isnull=True, created_at__lte=delete_period)
-    )
-    payloads = EventPayload.objects.filter(
-        deliveries__isnull=True, created_at__lte=delete_period
-    )
-
-    attempts._raw_delete(attempts.db)  # type: ignore[attr-defined] # raw access # noqa: E501
-    deliveries._raw_delete(deliveries.db)  # type: ignore[attr-defined] # raw access # noqa: E501
-    payloads._raw_delete(payloads.db)  # type: ignore[attr-defined] # raw access # noqa: E501
+    valid_deliveries = EventDelivery.objects.filter(created_at__gt=delete_period)
+    payloads_to_delete = EventPayload.objects.filter(
+        ~Exists(valid_deliveries.filter(payload_id=OuterRef("id")))
+    ).order_by("-pk")
+    ids = payloads_to_delete.values_list("pk", flat=True)[:BATCH_SIZE]
+    qs = EventPayload.objects.filter(pk__in=ids)
+    if ids:
+        qs.delete()
+        delete_event_payloads_task.delay()
```

Comment:
Same comment as here: https://github.com/saleor/saleor/pull/13246#discussion_r1258298416

Reply:
We have decided that we should check all our tasks in search of such probability and make a separate PR with fix that will prevent that from happening in all of them. So that will be out of scope of this PR.

Reply:
Why not fix it now instead of introducing technical debt in a bug fix? This would be highly problematic if we start queuing tons of concurrent writes on the same rows as it is a "recursive" function.

You can do something similar as here: https://github.com/saleor/saleor/blob/5e7c57dad9e64b09477ebdcee53f0277359bc598/saleor/checkout/tasks.py#L86-L96

Reply:
You could also set an expiration date/timestamp and pass it to the task. If null then you need to give one to the sub tasks, otherwise you just check the value is lower than current time and pass it along the value.

Reply:
@NyanKiyoshi I've made changes. What do you think about it?

Reply:
Looks good! 👌